rpcserver.go

```diff
@@ -6777,7 +6778,35 @@ func (r *rpcServer) ForwardingHistory(ctx context.Context,
 	}
 	timeSlice, err := r.server.miscDB.ForwardingLog().Query(eventQuery)
 	if err != nil {
-		return nil, fmt.Errorf("unable to query forwarding log: %v", err)
+		return nil, fmt.Errorf("unable to query forwarding log: %v",
+			err)
+	}
+
+	// Helper function to extract a peer's node alias given its SCID.
+	getRemoteAlias := func(chanId lnwire.ShortChannelID) string {
+		edge, err := r.GetChanInfo(ctx, &lnrpc.ChanInfoRequest{
+			ChanId: chanId.ToUint64(),
+		})
+		if err != nil {
+			return ""
+		}
+
+		remotePub := ""
+		switch r.selfNode.String() {
+		case edge.Node1Pub:
+			remotePub = edge.Node2Pub
+		case edge.Node2Pub:
+			remotePub = edge.Node1Pub
+		}
+
+		info, err := r.GetNodeInfo(ctx, &lnrpc.NodeInfoRequest{
```

Comment:
We should use `r.server.graphDB.FetchLightningNode()` directly here, because `r.GetNodeInfo()` also loads all channels of a node which can be quite expensive.
Looking at the `GetNodeInfo` RPC, we maybe should add a flag (in another PR) like `no_channel_stats` that skips the channel lookup to make the call faster. Otherwise if a UI application fetches the info for each peer of each channel this can be quite a load on the DB.

Reply:
Thank you for the review and the performance consideration. 

Is performance of `GetNodeInfo` still a concern to you if we exclude loading the channels?
It would be a better user XP if the peer alias names could be included without an additional flag.

Reply:
Thanks for the review and pointing to performance. 

Just to confirm with you - Is performance still an issue if we use `r.server.graphDB.FetchLightningNode()` here?
I am thinking that it would be nice from a user experience perspective if peer alias names could be included without an additional flag so want to check if we can omit that.

Reply:
Using `FetchLightningNode` directly will reduce the number of DB calls drastically, but there will still be at least two calls per forwarded payment. Having the information by default sounds nice though. So how about we do the opposite? Something like `--skip_alias_lookup` for large nodes that have too many forwards to do the lookup in reasonable time?

Reply:
Yes, that's a good idea to give the option to omit the the lookup! I will add that, thx.